### PR TITLE
Add jackknifing

### PR DESCRIPTION
**Pull request by tensionhead**
Changes Summary
----------------------------
- added jackknife recipe for leave-one-out trial averages
- included tests with application for coherence

Still To Do
---------------
- wire it into the connectivityanalysis frontend

Reviewer Checklist
------------------
- [x] Are testing routines present?
- [x] Do objects in the global package namespace perform proper parsing of their input? 
- [x] Are all docstrings complete and accurate?
- [x] Is the CHANGELOG.md up to date?

### CODE REVIEW

**Comment by dfsp-spirit** [comment]
Please add this great new feature to the Changelog.

**Comment by tensionhead** [comment]
I would wait for the Changelog until some functionality is exposed to the user API?!

**Comment by dfsp-spirit** [comment]
> 



> I would wait for the Changelog until some functionality is exposed to the user API?!

Okay!

**Review by dfsp-spirit — CHANGES_REQUESTED** [review]
Thanks a lot, I just left some minor comments on some technical stuff (some unused vars/imports). Otherwise I think this is ready to be merged.

File: syncopy/tests/test_basedata.py [hunk 1]
```diff
@@ -93,12 +92,6 @@ def test_data_alloc(self):
                 assert dummy.mode == "r+", dummy.data.file.mode
                 del dummy
 
-                # allocation using array + filename
-                dummy = getattr(spd, dclass)(data=self.data[dclass], filename=fname)
-                assert dummy.filename == fname
-                assert np.array_equal(dummy.data, self.data[dclass])
-                del dummy
-
```
[the remark this answers is not in the document]
Comment (tensionhead) [reply syncopy/tests/test_basedata.py:101]: I very much think so yes..

File: syncopy/statistics/jackknifing.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+    with h5py.File(replicates._filename, mode='w') as h5file:
+        dset = h5file.create_dataset('data', shape=trl_ensemble.data.shape,
+                                     dtype=trl_ensemble.data.dtype)
+        replicates.data = dset
+
+    # we still need to write into it
+    replicates._reopen()
+
+    log_dict = {}
```
[the remark this answers is not in the document]
Comment (tensionhead) [reply syncopy/statistics/jackknifing.py:94]: True, probably better to let the future frontend integration handle the log..

**Review by dfsp-spirit — APPROVED** [review]

**Pull request merged**